Compiler/BackEnd/Initialization.mo

```diff
@@ -1426,7 +1425,8 @@ algorithm
     outM := arrayCreate(inNEqns+inNAddEqns, {});
     outM := Array.copy(inM, outM);
     newEqIndices := List.intRange2(inNEqns+1, inNEqns+inNAddEqns);
-    outM := List.fold1(newEqIndices, squareIncidenceMatrix1, inInitVarIndices, outM);
+    allVarIndices := List.intRange2(1,inNEqns+inNAddEqns);
+    outM := List.fold1(newEqIndices, squareIncidenceMatrix1, allVarIndices, outM);
```

Comment:
This doesn't look right to me. The introduced artificial equations should only be connected to `inInitVarIndices` (states, discrete states, and parameters having fixed=false).